sdks/java/maven-archetypes/examples/src/main/resources/archetype-resources/pom.xml

```diff
@@ -132,6 +132,9 @@
               </filters>
               <transformers>
                 <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                <transformer implementation="org.apache.maven.plugins.shade.resource.AppendingTransformer">
+                  <resource>reference.conf</resource>
```

Comment:
Could you clarify where "reference.conf" comes from ? I'm not familiar regarding this file.

Reply:
Flink uses [akka](https://akka.io/), which in turn uses [typesafe config](https://github.com/lightbend/config) for configuration. The typesafe config uses two types of configuration files - `reference.conf` and `application.conf`, where the first holds defaults which the latter can (optionally) override. Both these files are located in resources of respective jar (library). Akka has several modules, each of which has its own `reference.conf`, which is fine as long as each of these jars is on classpath. When we shade them together, we must append the respective `reference.conf`s so that every default value of each (sub)module has a default value.

https://doc.akka.io/docs/akka/current/additional/packaging.html#maven-jarjar-onejar-or-assembly

Reply:
Thanks for clarifying.